fix:issue-60/tweets in threads

surajsharma opened the pull request:
- successfully migrated [twitter-widget](https://github.com/holdex/twitter-widget) component into the repo
- redesigned the tweet widget as per new design
- tweets are now rendering within threads
- ~~using a CORS proxy for fetching tweets as a temporary measure~~
- migrated css too tailwind where possible

fixes #60

georgeciubotaru commented:
@surajsharma Please see this tab, and try to check for unresolved comments: https://github.com/holdex/holdex-venture-studio/pull/317/files

surajsharma commented:
@georgeciubotaru done

georgeciubotaru commented on src/lib/components/BodyRenderer/Blocks/Tweet/components/twitter-layout/LinkPreview/style.sass:
```diff
@@ -0,0 +1,36 @@
+@use "sass:math"
+
+.embed-link
+    @apply relative inline-block w-full no-underline leading-none z-[1] mt-3 rounded-2xl bg-tw-tweet-bg-color border border-tw-tweet-border
+        transition: background-color 0.3s ease-in-out, border-color 0.3s ease-in-out
```
@surajsharma try to put the `@apply` as the last property, to avoid the wrong indentation

surajsharma replied:
@georgeciubotaru although i have put it as a last property, i think we can do away with it here altogether

georgeciubotaru commented on src/lib/components/BodyRenderer/Blocks/Tweet/components/twitter-layout/headings/permalink.svelte:
```diff
@@ -0,0 +1,25 @@
+
+<style lang="sass">
+
+    @media screen and (min-width: 992px) 
+    a 
+        @apply mr-2
```
indentation is wrong

georgeciubotaru commented:
@surajsharma Last comments

1. Still we have dev dependencies that we don't need
2. lib folder is present (we don't need it)
3. importing `theme.scss` in the Twitter components (we don't need it as it will inherit from the main layout)
4. See my comments above

surajsharma commented:
@georgeciubotaru 

1. removed unused dev deps, including all `@types`
2. deleted the said folder, sorry didn't feel confident deleting earlier
3. can you show me where this is in the code? i searched but found nothing, have deleted all of these
4. fixed

markholdex commented:
@surajsharma @georgeciubotaru what are we missing to complete this task?

surajsharma commented:
@markholdex all done at my end, waiting on @georgeciubotaru for the final word

georgeciubotaru commented on src/lib/components/BodyRenderer/Blocks/Tweet/components/twitter-layout/LinkPreview/style.sass:
```diff
@@ -0,0 +1,36 @@
+        bg-tw-tweet-bg-color border-b border-tw-tweet-border @apply
+
+    &__content
+        relative w-full block text-[15px] leading-5 whitespace-normal p-3 @apply
```
@surajsharma not like that
```
.class {
	// CSS declarations
	text-align: center
	@apply text-center
}
```
In that case the `CSS` declaration won't be indented  like that
```
.class {
	@apply text-center
		// CSS declarations
		text-align: center
}
```

surajsharma replied:
@georgeciubotaru fixed, pls see

georgeciubotaru commented on package.json:
```diff
@@ -14,49 +14,44 @@
 		"@steeze-ui/heroicons": "^2.2.2",
 		"@sveltejs/adapter-auto": "^2.0.0",
 		"@sveltejs/kit": "^1.5.0",
-		"@types/lodash-es": "^4.17.7",
```
@surajsharma add types back

georgeciubotaru commented on package.json:
```diff
@@ -14,49 +14,44 @@
 	},
-	"type": "module",
 	"dependencies": {
+		"@github/details-dialog-element": "^3.1.3",
+		"open-graph-scraper": "^4.11.0",
+		"twitter-text": "^3.1.0",
```
remove these ones

georgeciubotaru commented on src/lib/components/BodyRenderer/Blocks/Tweet/components/twitter-layout/Tweet/TweetAction.svelte:
```diff
@@ -0,0 +1,95 @@
+
+<style lang="sass">
+
+    @import "../../../../../../../styles/theme"
```
check for this import in all files.

You don't need it

georgeciubotaru reviewed (changes requested):
@surajsharma See my changes, iterate in all files and to the updates

georgeciubotaru commented on package.json:
```diff
@@ -14,41 +14,35 @@
 		"@sveltejs/adapter-auto": "^2.0.0",
 		"@sveltejs/kit": "^1.5.0",
 		"@types/lodash-es": "^4.17.7",
-		"@types/numeral": "^2.0.2",
-		"@types/rimraf": "^4.0.5",
```
these put back

georgeciubotaru commented on package.json:
```diff
@@ -14,41 +14,35 @@
-		"svelte-check": "^3.0.1",
-		"svelte-preprocess": "^5.0.3",
 		"tailwindcss": "^3.2.7",
-		"tslib": "^2.4.1",
```
@surajsharma see the removed deps and keep only the ones we are using

surajsharma replied:
@georgeciubotaru i search for `tslib` but no use found, closing...

georgeciubotaru commented on package.json:
```diff
@@ -14,41 +14,35 @@
 		"@apollo/client": "^3.7.10",
-		"@editorjs/editorjs": "^2.26.5",
 		"apexcharts": "^3.37.3",
-		"dotenv": "^16.0.3",
```
we need this

georgeciubotaru commented:
@surajsharma the last ones and is good to go.

Good job!

Hope we will minimize the back and forth in the future PR.

surajsharma commented:
@georgeciubotaru yes i look forward to smaller PRs :)

georgeciubotaru commented:
@surajsharma On It
